tests/test_views.py

```diff
@@ -202,26 +197,65 @@ def test_get_render_patched(
         for key, value in expected_context.items():
             assert response.context[key] == value  # pylint: disable=no-member
 
+    @mock.patch('enterprise.views.get_partial_pipeline')
+    @mock.patch('enterprise.views.lift_quarantine')
+    @mock.patch('enterprise.views.quarantine_session')
+    @mock.patch('enterprise.views.get_real_social_auth_object')
+    @mock.patch('enterprise.views.get_complete_url')
+    @mock.patch('enterprise.views.render', side_effect=fake_render)
+    @mock.patch('enterprise.views.get_enterprise_customer_for_request')
+    @mock.patch('enterprise.views.configuration_helpers')
+    def test_get_render_patched_optional(
+            self,
+            config_mock,
+            get_ec_mock,
+            *args
+    ):  # pylint: disable=unused-argument
+        """
+        Test that we have correct context for an optional form rendering.
+        """
+        config_mock.get_value.return_value = 'This Platform'
+        fake_ec = mock.MagicMock(
+            enforces_data_sharing_consent=mock.MagicMock(return_value=False)
+        )
+        fake_ec.name = 'Fake Customer Name'
+        get_ec_mock.return_value = fake_ec
+        client = Client()
+        response = client.get(self.url)
+        expected_prompt = CONSENT_REQUEST_PROMPT.format(  # pylint: disable=no-member
+            enterprise_customer_name=fake_ec.name
+        )
+        expected_alert = CONFIRMATION_ALERT_PROMPT.format(  # pylint: disable=no-member
+            enterprise_customer_name=fake_ec.name
+        )
+        expected_warning = CONFIRMATION_ALERT_PROMPT_WARNING.format(  # pylint: disable=no-member
+            enterprise_customer_name=fake_ec.name
+        )
+        expected_context = {
+            'consent_request_prompt': expected_prompt,
+            'confirmation_alert_prompt': expected_alert,
+            'confirmation_alert_prompt_warning': expected_warning,
+            'platform_name': 'This Platform',
+            'enterprise_customer_name': 'Fake Customer Name',
+        }
+        for key, value in expected_context.items():
+            assert response.context[key] == value  # pylint: disable=no-member
```

Comment:
The above function is the only significant change in this file -- the rest is just adding some necessary mock patches and refactoring to make the file a little shorter and more consistent.